homeassistant/components/senseme/fan.py

```diff
@@ -0,0 +1,125 @@
+"""Support for Big Ass Fans SenseME fan."""
+from __future__ import annotations
+
+import math
+from typing import Any
+
+from aiosenseme import SensemeFan
+
+from homeassistant import config_entries
+from homeassistant.components.fan import (
+    DIRECTION_FORWARD,
+    DIRECTION_REVERSE,
+    SUPPORT_DIRECTION,
+    SUPPORT_SET_SPEED,
+    FanEntity,
+)
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.util.percentage import (
+    percentage_to_ranged_value,
+    ranged_value_to_percentage,
+)
+
+from .const import (
+    DOMAIN,
+    PRESET_MODE_WHOOSH,
+    SENSEME_DIRECTION_FORWARD,
+    SENSEME_DIRECTION_REVERSE,
+)
+from .entity import SensemeEntity
+
+SENSEME_DIRECTION_TO_HASS = {
+    SENSEME_DIRECTION_FORWARD: DIRECTION_FORWARD,
+    SENSEME_DIRECTION_REVERSE: DIRECTION_REVERSE,
+}
+HASS_DIRECTION_TO_SENSEME = {v: k for k, v in SENSEME_DIRECTION_TO_HASS.items()}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: config_entries.ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up SenseME fans."""
+    device = hass.data[DOMAIN][entry.entry_id]
+    if device.is_fan:
+        async_add_entities([HASensemeFan(device)])
+
+
+class HASensemeFan(SensemeEntity, FanEntity):
+    """SenseME ceiling fan component."""
+
+    _attr_supported_features = SUPPORT_SET_SPEED | SUPPORT_DIRECTION
+    _attr_preset_modes = [PRESET_MODE_WHOOSH]
+
+    def __init__(self, device: SensemeFan) -> None:
+        """Initialize the entity."""
+        super().__init__(device, device.name)
+        self._attr_speed_count = self._device.fan_speed_max
+        self._attr_unique_id = f"{self._device.uuid}-FAN"  # for legacy compat
+        self._async_update_attrs()
+
+    @callback
+    def _async_update_attrs(self) -> None:
+        """Update attrs from device."""
+        self._attr_is_on = self._device.fan_on
+        self._attr_current_direction = SENSEME_DIRECTION_TO_HASS.get(
+            self._device.fan_dir, DIRECTION_FORWARD  # None also means forward
+        )
+        if self._device.fan_speed is not None:
+            self._attr_percentage = ranged_value_to_percentage(
+                self._device.fan_speed_limits, self._device.fan_speed
+            )
+        else:
+            self._attr_percentage = None
+        whoosh = self._device.fan_whoosh_mode
+        self._attr_preset_mode = whoosh if whoosh else None
+        super()._async_update_attrs()
+
+    @property
+    def extra_state_attributes(self) -> dict:
+        """Get the current device state attributes."""
+        return {
+            "auto_comfort": self._device.fan_autocomfort.capitalize(),
+            "smartmode": self._device.fan_smartmode.capitalize(),
```

Comment:
What do the values represent for these state attributes?

Reply:
Auto Comfort allows the fan to monitor and adjust to room conditions like temperature, humidity and occupancy. There are four possible states: "Off", "Cooling", "Heating", and "Followtstat".

Smartmode indicates the fan's comfort mode. When auto_comfort is set to "Followtstat" the actual auto_comfort value will change based the connected thermostat otherwise smartmode tracks auto_comfort. There are three possible states: "Off", "Cooling" and "Heating".

There is currently no support for controlling these. It is likely they could be moved to the `select` platform in a future PR.

Reply:
Ok, thanks.